`RequestListenersTest.testCompletionListeners()` is flaky

**Jetty version(s)**
12.0.x

**Description**
`RequestListenersTest.testCompletionListeners()` sometimes fails with:

```
java.lang.AssertionError: 
Expected: iterable containing ["zero", "one", "two", "three", "four"]
     but: 

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.eclipse.jetty.server/org.eclipse.jetty.server.RequestListenersTest.testCompletionListeners(RequestListenersTest.java:115)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
```

See: https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-11627/16/tests

--- a/jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/RequestListenersTest.java
+++ b/jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/RequestListenersTest.java
@@ -37,6 +37,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 
+import static org.awaitility.Awaitility.await;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsString;
@@ -112,7 +113,7 @@ public class RequestListenersTest
             """));
 
         assertEquals(HttpStatus.OK_200, response.getStatus());
-        assertThat(history, contains("zero", "one", "two", "three", "four"));
+        await().atMost(5, TimeUnit.SECONDS).until(() -> history, contains("zero", "one", "two", "three", "four"));
     }
 
     @Test

